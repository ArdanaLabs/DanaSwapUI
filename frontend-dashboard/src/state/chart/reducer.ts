import {
  ActionReducerMapBuilder,
  PayloadAction,
  createReducer,
} from "@reduxjs/toolkit"

import * as E from "fp-ts/Either"
import { RemoteData, fromEither, pending } from "fp-ts-remote-data"

import {
  receivedAggVolume,
  receivedAggLiquidity,
  receivedPoolFees,
  receivedPoolVolume,
  receivedPoolLiquidity,
  receivedPoolTxCount,
  receivedPoolAPY,
  receivedPoolTransactions,
} from "./actions"

import {
  APYChart,
  FeeVolumeChart,
  LiquidityChart,
  TxCountChart,
  VolumeChart,
} from "Data/Chart"
import { FetchDecodeError, FetchDecodeResult } from "Data/FetchDecode"
import * as Transaction from "Data/Transaction"

export type State = {
  aggLiquidity: RemoteData<FetchDecodeError, LiquidityChart.Type>
  aggVolume: RemoteData<FetchDecodeError, VolumeChart.Type>
  poolFees: RemoteData<FetchDecodeError, FeeVolumeChart.Type>
  poolVolume: RemoteData<FetchDecodeError, VolumeChart.Type>
  poolLiquidity: RemoteData<FetchDecodeError, LiquidityChart.Type>
  poolTxCount: RemoteData<FetchDecodeError, TxCountChart.Type>
  poolAPY: RemoteData<FetchDecodeError, APYChart.Type>
  poolTransactions: RemoteData<FetchDecodeError, Transaction.WithTotalValue[]>
}

export const initialState: State = {
  aggLiquidity: pending,
  aggVolume: pending,
  poolFees: pending,
  poolVolume: pending,
  poolLiquidity: pending,
  poolTxCount: pending,
  poolAPY: pending,
  poolTransactions: pending,
}

// side-effectful: function that logs on Left, and returns the typical fromEither
// TODO: move this to chain with the decoder stuff
function fromEitherLogFail<A>({
  payload,
}: PayloadAction<FetchDecodeResult<A>>): RemoteData<FetchDecodeError, A> {
  if (E.isLeft(payload)) {
    console.error("Fetch error:", payload.left)
  }
  return fromEither(payload)
}

export default createReducer(
  initialState,
  (builder: ActionReducerMapBuilder<State>) =>
    builder
      .addCase(receivedAggLiquidity, (state, action) => ({
        ...state,
        aggLiquidity: fromEitherLogFail(action),
      }))
      .addCase(receivedAggVolume, (state, action) => ({
        ...state,
        aggVolume: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolFees, (state, action) => ({
        ...state,
        poolFees: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolVolume, (state, action) => ({
        ...state,
        poolVolume: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolLiquidity, (state, action) => ({
        ...state,
        poolLiquidity: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolTxCount, (state, action) => ({
        ...state,
        poolTxCount: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolAPY, (state, action) => ({
        ...state,
        poolAPY: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolTransactions, (state, action) => ({
        ...state,
        poolTransactions: fromEitherLogFail(action),
      }))
)
