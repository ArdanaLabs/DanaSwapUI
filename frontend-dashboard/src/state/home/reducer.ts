import {
  ActionReducerMapBuilder,
  PayloadAction,
  createReducer,
} from "@reduxjs/toolkit"

import { RemoteData, fromEither, pending } from "fp-ts-remote-data"

import { CombinedStats } from "Data/Stats/CombinedStats"
import { FetchDecodeError, FetchDecodeResult } from "Data/FetchDecode"

import { receivedCombinedStats } from "./actions"

export type State = RemoteData<FetchDecodeError, CombinedStats>

export const initialState = pending

export default createReducer(
  initialState,
  (builder: ActionReducerMapBuilder<State>) =>
    builder.addCase(
      receivedCombinedStats,
      (
        _state: State,
        action: PayloadAction<FetchDecodeResult<CombinedStats>>
      ) => fromEither(action.payload)
    )
)
