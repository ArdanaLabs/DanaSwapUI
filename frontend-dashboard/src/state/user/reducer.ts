import { ActionReducerMapBuilder, createReducer } from "@reduxjs/toolkit"

import * as O from "fp-ts/Option"

import { SupportedW3ColorScheme, Theme } from "Data/User/Theme"

import { updatePrefersColorScheme, updateUserTheme } from "./actions"

function currentTimestamp(): number {
  return new Date().getTime()
}

export const THEME_LOCALSTORAGE_KEY = "user/theme"

export interface UserState {
  lastUpdateVersionTimestamp?: number
  theme: O.Option<Theme> // current theme
  prefersColorScheme: O.Option<SupportedW3ColorScheme>
  timestamp: number
}

export const initialState: UserState = {
  theme: O.none,
  prefersColorScheme: O.none,
  timestamp: currentTimestamp(), // side-effect
}

export default createReducer(
  initialState,
  (builder: ActionReducerMapBuilder<UserState>) =>
    builder
      .addCase(updateUserTheme, (state, action) => {
        localStorage.setItem(
          THEME_LOCALSTORAGE_KEY,
          action.payload.toString(10)
        )
        const timestamp = currentTimestamp()
        return { ...state, theme: O.some(action.payload), timestamp }
      })
      .addCase(updatePrefersColorScheme, (state, action) => {
        const timestamp = currentTimestamp()
        return { ...state, prefersColorScheme: action.payload, timestamp }
      })
)
