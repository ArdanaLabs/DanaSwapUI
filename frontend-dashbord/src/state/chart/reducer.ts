import { createReducer } from "@reduxjs/toolkit";

import { updateAggVol } from "./actions";

export const initialState: any = {
  aggVolsPerFiveMins: [],
};

export default createReducer(initialState, (builder) =>
  builder.addCase(updateAggVol, (state, action) => {
  })
);
