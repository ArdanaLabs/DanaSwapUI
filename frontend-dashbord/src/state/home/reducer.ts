import { createReducer } from "@reduxjs/toolkit";

import { TotalStat, updateTotalStats } from "./actions";

export const initialState: TotalStat = {
  totalDepositsAllPoolsUSD: 0,
  totalDailyVolumeUSD: null,
  totalDailyTxCount: 0,
  totalDailyFeeVolumeUSD: 0,
  totalLiquidityUtilization: 0,
  poolStats: null,
};

export default createReducer(initialState, (builder) =>
  builder.addCase(updateTotalStats, (state, action) => {
    const {
      totalDepositsAllPoolsUSD,
      totalDailyVolumeUSD,
      totalDailyTxCount,
      totalDailyFeeVolumeUSD,
      totalLiquidityUtilization,
      poolStats,
    } = action.payload;

    state.totalDepositsAllPoolsUSD = totalDepositsAllPoolsUSD;
    state.totalDailyVolumeUSD = totalDailyVolumeUSD;
    state.totalDailyTxCount = totalDailyTxCount;
    state.totalDailyFeeVolumeUSD = totalDailyFeeVolumeUSD;
    state.totalLiquidityUtilization = totalLiquidityUtilization;

    state.poolStats = poolStats;
  })
);
