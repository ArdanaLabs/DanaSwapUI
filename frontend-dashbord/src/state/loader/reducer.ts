import { createReducer } from "@reduxjs/toolkit";

import { setLoading } from "./actions";

export const initialState = {
  loading: false
};

export default createReducer(initialState, (builder) =>
  builder.addCase(setLoading, (state, action) => {
    state.loading = action.payload;
  })
);
