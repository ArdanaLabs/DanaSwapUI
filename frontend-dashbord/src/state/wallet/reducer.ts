import { createReducer } from "@reduxjs/toolkit";

import { getAccountKeys } from "./actions";

export interface WalletStateType {
  privateKey: string | null;
  publicKey: string | null;
  rewardAddress: string | null;
  accountId: string | null;
  accountIdFull: string | null;
}

export const initialState: WalletStateType = {
  privateKey: null,
  publicKey: null,
  rewardAddress: null,
  accountId: null,
  accountIdFull: null,
};

export default createReducer(initialState, (builder) =>
  builder.addCase(getAccountKeys, (state, action) => {
    state.privateKey = action.payload.privateKey;
    state.publicKey = action.payload.publicKey;
    state.rewardAddress = action.payload.rewardAddress;
    state.accountId = action.payload.accountId;
    state.accountIdFull = action.payload.accountIdFull;
  })
);
