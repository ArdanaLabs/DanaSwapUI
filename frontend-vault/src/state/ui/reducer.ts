import { createReducer } from "@reduxjs/toolkit"
import { toggleUiModalAction } from "./actions"

export interface UiState {
  modal: {
    open: boolean
    type?: string
  }
}

export const initialState: UiState = {
  modal: {
    open: false,
    type: "",
  },
}

export default createReducer(initialState, (builder) =>
  builder.addCase(toggleUiModalAction, (state, action) => {
    state.modal = { ...state.modal, ...action.payload }
  })
)
