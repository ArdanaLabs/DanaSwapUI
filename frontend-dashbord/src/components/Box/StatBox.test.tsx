import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import * as Enzyme from "enzyme";
import configureMockStore, { MockStoreEnhanced } from "redux-mock-store";
import { shallow } from "enzyme";
import { Provider, useSelector } from "react-redux";
import { initialState } from "state/user/reducer";
import { StatBox } from ".";

Enzyme.configure({ adapter: new Adapter() });

const mockDispatch = jest.fn();
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useEffect: jest.fn((f) => f()),
}));
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));
const mockStore = configureMockStore([]);

let store: MockStoreEnhanced<unknown>;

describe("Components / Box / StatBox", () => {
  beforeEach(() => {
    store = mockStore(initialState);
    (useSelector as jest.Mock).mockImplementation((callback) => {
      return callback(initialState);
    });
  });

  afterEach(() => {
    (useSelector as jest.Mock).mockClear();
  });

  it("renders", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <StatBox
          image={""}
          title="TOTAL VALUE LOCKED"
          content="$1,234,567"
          delay={0}
        />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
