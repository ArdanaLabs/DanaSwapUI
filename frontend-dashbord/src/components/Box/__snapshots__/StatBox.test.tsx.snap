// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Box / StatBox renders 1`] = `ShallowWrapper {}`;
